File: backend/internal/domain/topic.go
package domain

import "time"

type Topic struct {
	ID          int       `json:"id"`
	CourseID    int       `json:"course_id"`
	Title       string    `json:"title"`
	Description string    `json:"description"`
	OrderNumber int       `json:"order_number"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

type TopicWithDetails struct {
	Topic
	CourseName    string       `json:"course_name"`
	Sessions      []Session    `json:"sessions"`
	Materials     []Material   `json:"materials"`
	Photos        []TopicPhoto `json:"photos"`
	SessionCount  int          `json:"session_count"`
	MaterialCount int          `json:"material_count"`
}

type TopicPhoto struct {
	ID        int       `json:"id"`
	TopicID   int       `json:"topic_id"`
	URL       string    `json:"url"`
	Notes     string    `json:"notes,omitempty"`
	CreatedAt time.Time `json:"created_at"`
}

type TopicWithSessions struct {
	Topic
	Sessions []Session `json:"sessions"`
}

File: frontend/src/types/topic.ts
export interface Topic {
  id: number;
  course_id: number;
  title: string;
  description: string;
  order_number: number;
  created_at: string;
  updated_at: string;
}

export interface TopicWithSessions extends Topic {
  sessions: TopicSession[];
}

export interface TopicWithDetails extends Topic {
  sessions: TopicSession[];
  materials: TopicMaterial[];
  session_count: number;
  material_count: number;
}

export interface TopicSession {
  id: number;
  course_id: number;
  number: number;
  title: string;
  description?: string;
  topic_id?: number;
  materials?: TopicMaterial[];
}

export interface TopicMaterial {
  id: number;
  session_id?: number;
  topic_id?: number;
  title: string;
  description?: string;
  type: string;
  url: string;
  created_at: string;
  updated_at: string;
}

export interface CreateTopicRequest {
  name: string;
  description?: string;
}

export interface UpdateTopicRequest {
  name?: string;
  description?: string;
}

export interface ReorderTopicsRequest {
  topic_ids: number[];
}
